assets/wizards/engagement/components/prompt.js

```diff
@@ -0,0 +1,294 @@
+					setSuccess( __( 'Prompt saved.', 'newspack' ) );
+					setIsDirty( false );
+					res();
+				} )
+				.catch( err => {
+					setError( err );
+					rej( err );
+				} )
+				.finally( () => {
+					setInFlight( false );
+				} );
+		} );
+	};
+
+	return (
+		<ActionCard
+			isMedium
+			expandable
+			collapse={ prompt.ready }
+			title={ prompt.title }
+			description={ sprintf(
+				// Translators: Status of the prompt.
+				__( 'Status: %s', 'newspack' ),
+				prompt.ready ? __( 'Ready', 'newspack' ) : __( 'Pending', 'newspack' )
+			) }
+			checkbox={ prompt.ready ? 'checked' : 'unchecked' }
+		>
+			{
+				<Grid columns={ 2 } gutter={ 16 } className="newspack-ras-campaign__grid">
+					<div className="newspack-ras-campaign__fields">
+						{ prompt.user_input_fields.map( field => (
+							<Fragment key={ field.name }>
+								{ 'array' === field.type && Array.isArray( field.options ) && (
+									<BaseControl
+										id={ `newspack-engagement-wizard__${ field.name }` }
+										label={ field.label }
+									>
+										{ field.options.map( option => (
+											<BaseControl
+												key={ option.id }
+												id={ `newspack-engagement-wizard__${ option.id }` }
+												className="newspack-checkbox-control"
+												help={ option.description }
+											>
+												<CheckboxControl
+													disabled={ inFlight }
+													label={ option.label }
+													value={ option.id }
+													checked={ values[ field.name ]?.indexOf( option.id ) > -1 }
+													onChange={ value => {
+														const toUpdate = { ...values };
+														if ( ! value && toUpdate[ field.name ].indexOf( option.id ) > -1 ) {
+															toUpdate[ field.name ].value = toUpdate[ field.name ].splice(
+																toUpdate[ field.name ].indexOf( option.id ),
+																1
+															);
+															setIsDirty( true );
+														}
+														if ( value && toUpdate[ field.name ].indexOf( option.id ) === -1 ) {
+															toUpdate[ field.name ].push( option.id );
+															setIsDirty( true );
+														}
+														setValues( toUpdate );
+													} }
+												/>
+											</BaseControl>
+										) ) }
+									</BaseControl>
+								) }
+								{ 'string' === field.type && 100 < field.max_length && (
+									<TextareaControl
+										className="newspack-textarea-control"
+										label={ field.label }
+										disabled={ inFlight }
+										help={ `${ values[ field.name ]?.length || 0 } / ${ field.max_length }` }
+										onChange={ value => {
+											if ( value.length > field.max_length ) {
+												return;
+											}
+
+											const toUpdate = { ...values };
+											toUpdate[ field.name ] = value;
+											if ( JSON.stringify( toUpdate ) !== JSON.stringify( values ) ) {
+												setIsDirty( true );
+											}
+											setValues( toUpdate );
+										} }
+										placeholder={ field.default }
```

Comment:
Being a default, shouldn't this be applied as the input value? I'm unable to save if I just want to use defaults but if I enter a field value and erase it (leaving it empty) I'm then able to save, which leads to another issue:

If the prompt is saved with only defaults (empty values) it remains pending and it looks like it's not stored in the database, even though the UI doesn't show errors and the API returns 200.

Reply:
Yes, I think your suggestion is closer to the spirit of @evilluendas's design. I updated this in 4f38c2d so that the fields are pre-populated with the default values.

[bdfde3e](https://github.com/Automattic/newspack-popups/pull/1090/commits/bdfde3ea9e66525301d5ffdec97c49b4377ae1bd) in Newspack Campaigns also addresses the second part of your comment: now, if you save a prompt without all of the required inputs, the inputs are saved but the prompt remains in a non-ready state and a warning message is shown stating that all required fields must be filled out to proceed.

Reply:
I tried saving a prompt without changing the defaults and I'm getting this message:

<img width="572" alt="image" src="https://user-images.githubusercontent.com/820752/232075998-4cdce2d8-74e4-4e8c-bac9-3f0db232b3bd.png">

I'm confused by its meaning, is it supposed to be an error?

Reply:
> I think it's easier just not to allow saving when a required field is missing
> I also think that leaving all newsletters on is better than none

Handled both of these in [15f97b4](https://github.com/Automattic/newspack-popups/pull/1090/commits/15f97b4098c454bf0098f8830980b0ff7fca0419). Newsletter lists enabled in ESP settings are selected by default, and improved the error messaging (and prevent saving) when required fields are missing: 

<img width="484" alt="Screen Shot 2023-04-14 at 12 33 01 PM" src="https://user-images.githubusercontent.com/2230142/232128283-a751a786-92c7-45d7-9db9-7ec8382dc58f.png">